### DISCUSSION TITLE
SO_SNDBUF 和 SO_RCVBUF 选项的疑问

### DISCUSSION THREAD

**Discussion by pexcn** [shadowsocks/shadowsocks-rust#486] [Q&A] [answered]
```bash
--inbound-recv-buffer-size
--inbound-send-buffer-size
--outbound-recv-buffer-size
--outbound-send-buffer-size
```
请问这些选项的推荐值是多少呢？应该如何设置？单位是 KB 还是 Byte ?

**Comment by zonyitoo** [comment, marked as answer]
Just don't set it if you don't know how exactly they work. It doesn't need to be set for normal applications.

**Reply by pexcn** [reply]
Thanks for your prompt reply.
According to https://github.com/shadowsocks/shadowsocks-libev/pull/2781, it seems to improve performance when it is set correctly, and I have some pursuit of the performance. Can you briefly describe how it is adjusted?

**Reply by zonyitoo** [reply]
They are corresponding to `SO_SNDBUF` and `SO_RCVBUF`. You can Google these two options and you will find lots of references.

**Reply by pexcn** [reply]
Thanks.
Then if I want to set to 8kb, should I use `--inbound-recv-buffer-size 8k` or `--inbound-recv-buffer-size 8192` ?

**Reply by zonyitoo** [reply]
`--inbound-recv-buffer-size 8192`

**Comment by database64128** [comment]
You really shouldn’t mess with these settings. In fact, these options shouldn’t have been added in the first place. Having them will only lead to misconfiguration by user, and messy, untidy code where bugs can hide.
